include/ccf/pal/snp_ioctl5.h

```diff
@@ -106,8 +106,16 @@ namespace ccf::pal::snp::ioctl5
       int fd = open(DEVICE, O_RDWR | O_CLOEXEC);
       if (fd < 0)
       {
-        throw std::logic_error(fmt::format("Failed to open \"{}\"", DEVICE));
+        throw std::logic_error(
+          fmt::format("Failed to open \"{}\" ({})", DEVICE, fd));
       }
+      auto close_fd = [](int* fd) {
```

Comment:
seems like a little function that can live in nonstd or somewhere rather than be duplicated, and we can probably name the pointer type too.